### PR TITLE
feat: design kit catalog

### LINKED ISSUE #919: Design kit catalog page
Spec: https://www.figma.com/file/I4Hryi0b79gU469AcZ3gkm/V1-Design-kit-Catalog?node-id=0%3A1

**Design**

- [x] External link redlines @kingtraceyj 
- [x] Specify body copy @kingtraceyj @aubrey-oneal 
- [x] Adobe Ace icon @kingtraceyj 

**Dev**

- [x] Route: `/design-kits`
- [x] Create reusable design kit tags in the data folder - [Matt's notes](https://ibm-studios.slack.com/archives/C035K6YM34Z/p1661974948885219)
- [x] Description that includes a link to the design getting started page
- [x] Search
- [x] Filter by maintainer, tool, type, status
- [x] Results bar
- [x] Catalog items
- [x] Pagination
- [x] External link hover treatment
- [ ] Make sure this link is added to the `Designing / get-started.mdx` line `184`

**Preview**

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #919 

Design Kit Catalog Page 🎉 

Make sure functionality and design is as expected in the redlines: 
https://www.figma.com/file/I4Hryi0b79gU469AcZ3gkm/RL_DesignKits_Catalog?node-id=3%3A1205

## Testing

- http://localhost:3000/design-kits

### CODE REVIEW

File: services/web-app/components/design-kit-catalog-item/design-kit-catalog-item-meta/design-kit-catalog-item-meta.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { Scales } from '@carbon/react/icons'
```
Comment (francinelucca) [inline services/web-app/components/design-kit-catalog-item/design-kit-catalog-item-meta/design-kit-catalog-item-meta.js:7]: this seems very similar to `AssetCatalogItemMeta`, is there an opportunity to convert that to `CatalogItemMeta` and reuse it?

File: services/web-app/components/design-kits-catalog/design-kits-catalog.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+import DesignKitCatalogItem from '@/components/design-kit-catalog-item'
+import { getDesignKitFilters } from '@/data/filters'
+import { sortItems } from '@/data/sort'
+import { designKitSortComparator } from '@/utils/schema'
+import { getSlug } from '@/utils/slug'
+import useQueryState from '@/utils/use-query-state'
+
+/**
+ * Returns true if a designKit should be included in the catalog results given the filter.
+ * @param {import('../../typedefs').DesignKit} designKit
```
Comment (francinelucca) [inline services/web-app/components/design-kits-catalog/design-kits-catalog.js:22]:
```suggestion
 * @param {import('@/typedefs').DesignKit} designKit
```

File: services/web-app/components/design-kits-catalog/design-kits-catalog.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  for (const [key, value] of Object.entries(filter)) {
+    if (!value.includes(designKit[key])) return false
+  }
+  return true
+}
+
+/**
+ * Finds and returns designKits that match a search criteria (name or description)
+ * from an array of designKits
+ * @param {import('../../typedefs').DesignKit[]} designKits list of designKits to filter
```
Comment (francinelucca) [inline services/web-app/components/design-kits-catalog/design-kits-catalog.js:37]:
```suggestion
 * @param {import('@/typedefs').DesignKit[]} designKits list of designKits to filter
```

File: services/web-app/components/design-kits-catalog/design-kits-catalog.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+ * @param {string} search search string to match designKits against
+ * @returns {import('../../typedefs').DesignKit[]} array of designKits that match search criteria
```
Comment (francinelucca) [inline services/web-app/components/design-kits-catalog/design-kits-catalog.js:39]:
```suggestion
 * @returns {import('@/typedefs').DesignKit[]} array of designKits that match search criteria
```

File: services/web-app/components/design-kits-catalog/design-kits-catalog.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    }
+    return true
+  })
+}
+
+/**
+ * Sorts and filters an array of designKits given a filter, sort key, and search query. Until a
+ * better solution is in place, the search is simply a filter to remove designKits that don't
+ * match any part of the name or description.
+ * @param {import('../../typedefs').DesignKit[]} designKits list of designKits to filter
```
Comment (francinelucca) [inline services/web-app/components/design-kits-catalog/design-kits-catalog.js:59]:
```suggestion
 * @param {import('@/typedefs').DesignKit[]} designKits list of designKits to filter
```

File: services/web-app/components/design-kits-catalog/design-kits-catalog.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+ * @param {object} filter filter object to apply to designKits
+ * @param {string} sort sort type to apply to designKits
+ * @param {string} search search string to match designKits against
+ * @returns {import('../../typedefs').DesignKit[]} array of filtered designKits
```
Comment (francinelucca) [inline services/web-app/components/design-kits-catalog/design-kits-catalog.js:63]:
```suggestion
 * @returns {import(@/typedefs').DesignKit[]} array of filtered designKits
```

File: services/web-app/components/design-kits-catalog/design-kits-catalog.js [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+ */
+const getFilteredDesignKits = (designKits, filter, sort, search) => {
+  const designKitsWithAppliedFilter = (designKits || []).filter((designKit) => {
+    return designKitIsInFilter(designKit, filter)
+  })
```
Comment (francinelucca) [inline services/web-app/components/design-kits-catalog/design-kits-catalog.js:68]:
```suggestion
const getFilteredDesignKits = (designKits, filter, sort, search) => {
  const designKitsWithAppliedFilter = (designKits || []).filter((designKit) => designKitIsInFilter(designKit, filter))
```

File: services/web-app/components/design-kits-catalog/design-kits-catalog.js [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+  )
+}
+
+export default DesignKitsCatalog
+
+DesignKitsCatalog.propTypes = {
+  /**
+   * Collection of designKits to display.
+   */
+  designKits: PropTypes.array.isRequired
```
Comment (francinelucca) [inline services/web-app/components/design-kits-catalog/design-kits-catalog.js:194]:
```suggestion
  designKits: PropTypes.arrayOf(designKitPropTypes).isRequired
```

File: services/web-app/types.js [hunk 9]
```diff
@@ -69,6 +69,19 @@ export const libraryPropTypes = PropTypes.shape({
...
+  name: PropTypes.string.isRequired,
+  description: PropTypes.string.isRequired,
+  tool: PropTypes.oneOf(['adobe-ase', 'adobe-xd', 'axure', 'figma', 'invision-freehand', 'sketch']),
+  type: PropTypes.oneOf(['elements', 'guidelines', 'ui', 'wireframes']),
+  status: PropTypes.oneOf(['draft', 'experimental', 'stable', 'deprecated']),
+  url: PropTypes.string.isRequired,
+  action: PropTypes.oneOf(['download', 'link']),
+  license: PropTypes.oneOf(['apache-2.0', 'ibm-internal', 'mit']),
+  noIndex: PropTypes.bool
+})
```
Comment (francinelucca) [inline services/web-app/types.js:83]:
```suggestion
export const designKitPropTypes = PropTypes.shape({
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  tool: PropTypes.oneOf(['adobe-ase', 'adobe-xd', 'axure', 'figma', 'invision-freehand', 'sketch']),
  type: PropTypes.oneOf(['elements', 'guidelines', 'ui', 'wireframes']),
  status: PropTypes.oneOf(['draft', 'experimental', 'stable', 'deprecated']),
  url: PropTypes.string.isRequired,
  action: PropTypes.oneOf(['download', 'link']),
  license: PropTypes.oneOf(['apache-2.0', 'ibm-internal', 'mit']),
  noIndex: PropTypes.bool
})
```

File: services/web-app/utils/schema.js [hunk 10]
```diff
@@ -9,18 +9,18 @@ import { capitalCase } from 'change-case'
...
-
-import { getSlug } from './slug'
+import { getSlug } from '@/utils/slug'
 /**
- * Defines the sort order of assets by status
- * @param {import('@/typedefs').Asset} assetA
- * @param {import('@/typedefs').Asset} assetB
+ * Defines the sort order of an element by status
+ * @param {import('../typedefs').Element} elementA
+ * @param {import('../typedefs').Element} elementB
```
Comment (francinelucca) [inline services/web-app/utils/schema.js:18]:
```suggestion
 * @param {import('@/typedefs').Element} elementA
 * @param {import('@/typedefs').Element} elementB
```

File: services/web-app/utils/schema.js [hunk 11]
```diff
@@ -75,12 +94,12 @@ export const getAssetType = (asset) => {
 }
 
 /**
- * Gets the asset status string value. If not found, defaults to 'draft'.
- * @param {import('@/typedefs').Asset} asset
- * @returns {string} Asset status value
+ * Gets the element status string value. If not found, defaults to 'draft'.
+ * @param {import('../typedefs').Element} element
```
Comment (francinelucca) [inline services/web-app/utils/schema.js:98]:
```suggestion
 * @param {import('@/typedefs').Element} element
```

File: services/web-app/utils/schema.js [hunk 12]
```diff
@@ -137,6 +156,22 @@ export const getLicense = (asset) => {
   return license
 }
 
+/**
+ * Gets the license from a designKit
+ * @param {import('../typedefs').DesignKit} designKit
+ * @returns {string} License
+ */
+export const getDesignKitLicense = (designKit) => {
+  const defaultLicense =
```
Comment (francinelucca) [inline services/web-app/utils/schema.js:165]: not sure what's happening here 🤔 , what is the intent?

File: services/web-app/utils/schema.js [hunk 13]
```diff
@@ -137,6 +156,22 @@ export const getLicense = (asset) => {
...
+    designKit.license === 'ibm-internal'
+      ? 'IBM Internal'
+      : designKit.license === 'ibm-internal'
+        ? 'MIT'
+        : null
+  const { license = defaultLicense } = designKit.license ? designKit.license : designKit
```
Comment (francinelucca) [inline services/web-app/utils/schema.js:171]:
```suggestion
  const { license = defaultLicense } = designKit
```

**Review by francinelucca — DISMISSED** [review]
Looking really good! some code cleanup suggestions

File: services/web-app/utils/schema.js (hunk 12 above)
Comment (andreancardona) [reply services/web-app/utils/schema.js:165]: Ooops ovrersight!

File: services/web-app/components/design-kit-catalog-item/design-kit-catalog-item-meta/design-kit-catalog-item-meta.js (hunk 1 above)
Comment (andreancardona) [reply services/web-app/components/design-kit-catalog-item/design-kit-catalog-item-meta/design-kit-catalog-item-meta.js]: done!

File: services/web-app/components/framework-icon/framework-icon.js [hunk 14]
```diff
@@ -41,7 +41,7 @@ FrameworkIcon.propTypes = {
   className: PropTypes.string,
   /**
    * Framework key from the asset schema. See:
-   * https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md.
+   * https://bit.ly/3BPZoTw
```
Comment (alisonjoseph) [inline services/web-app/components/framework-icon/framework-icon.js:44]: Did you mean to change this? This seems to link to a google doc 😉

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
LGTM other than the random link change!

File: services/web-app/components/design-kit-icon/design-kit-icon.js [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+DesignKitIcon.propTypes = {
+  /**
+   * Optional container class name.
+   */
+  className: PropTypes.string,
+  /**
+   * Design kits key from the schema. See:
+   * https://bit.ly/3BPZoTw
```
Comment (alisonjoseph) [inline services/web-app/components/design-kit-icon/design-kit-icon.js:41]: I think this link might be wrong too?

File: services/web-app/components/asset-catalog-item/asset-catalog-item.js [hunk 16]
```diff
@@ -74,19 +75,28 @@ const AssetCatalogItemContent = ({ asset, isGrid = false, otherFrameworkCount =
...
         )}
         {!isSeparatedMeta && (
-          <AssetCatalogItemMeta
-            asset={asset}
+          <CatalogItemMeta
+            element={asset}
             className={styles['meta--absolute']}
             properties={['status', 'license']}
+            license={getAssetLicense(asset)}
+            statusKey={asset?.statusKey}
```
Comment (francinelucca) [inline services/web-app/components/asset-catalog-item/asset-catalog-item.js:99]:
```suggestion
          <CatalogItemMeta
            className={styles['meta--absolute']}
            properties={['status', 'license']}
            license={getAssetLicense(asset)}
            statusKey={asset?.statusKey}
```

File: services/web-app/components/catalog-item-meta/catalog-item-meta.js [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+        <li className={styles['meta-item']} key={i}>
+          {prop === 'status' && renderStatus()}
+          {prop === 'license' && renderLicense()}
+        </li>
+      ))}
+    </ul>
+  )
+}
+
+ElementCatalogItemMeta.propTypes = {
```
Comment (francinelucca) [inline services/web-app/components/catalog-item-meta/catalog-item-meta.js:53]:
```suggestion
CatalogItemMeta.propTypes = {
```

File: services/web-app/components/catalog-item-meta/catalog-item-meta.js [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+   * Element properties.
+   */
+  properties: PropTypes.array,
+  /**
+   * Element status key.
+   */
+  statusKey: PropTypes.array
+}
+
+export default ElementCatalogItemMeta
```
Comment (francinelucca) [inline services/web-app/components/catalog-item-meta/catalog-item-meta.js:72]:
```suggestion
export default CatalogItemMeta
```

File: services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+import Link from 'next/link'
+import { useState } from 'react'
+
+import { designKitTypes } from '@/data/design-kit-types'
+import { teams } from '@/data/teams'
+import { designKitPropTypes } from '@/types'
+import { getDesignKitLicense } from '@/utils/schema'
+import { mediaQueries, useMatchMedia } from '@/utils/use-match-media'
+
+import CatalogItemMeta from '../catalog-item-meta/index'
```
Comment (francinelucca) [inline services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js:19]:
```suggestion
import CatalogItemMeta from '../catalog-item-meta'
```

File: services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+                    {iconRender()}
+                  </div>
+                  {isSeparatedMeta && (
+                    <>
+                      <CatalogItemMeta
+                        element={designKit}
+                        properties={['license']}
+                        license={getDesignKitLicense(designKit)}
+                        statusKey={designKit?.statusKey}
+                      />
```
Comment (francinelucca) [inline services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js:80]:
```suggestion
                      <CatalogItemMeta
                        properties={['license']}
                        license={getDesignKitLicense(designKit)}
                        statusKey={designKit?.statusKey}
                      />
```

File: services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+                      <CatalogItemMeta
+                        element={designKit}
+                        className={styles['meta--absolute']}
+                        properties={['status']}
+                        license={getDesignKitLicense(designKit)}
+                        statusKey={designKit?.statusKey}
+                      />
```
Comment (francinelucca) [inline services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js:87]:
```suggestion
                      <CatalogItemMeta
                        className={styles['meta--absolute']}
                        properties={['status']}
                        license={getDesignKitLicense(designKit)}
                        statusKey={designKit?.statusKey}
                      />
```

File: services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+                    </>
+                  )}
+                  {!isSeparatedMeta && (
+                    <CatalogItemMeta
+                      element={designKit}
+                      className={styles['meta--absolute']}
+                      properties={['status', 'license']}
+                      license={getDesignKitLicense(designKit)}
+                      statusKey={designKit?.statusKey}
+                    />
```
Comment (francinelucca) [inline services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js:97]:
```suggestion
                    <CatalogItemMeta
                      className={styles['meta--absolute']}
                      properties={['status', 'license']}
                      license={getDesignKitLicense(designKit)}
                      statusKey={designKit?.statusKey}
                    />
```

File: services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js [hunk 23]
```diff
@@ -0,0 +1,126 @@
...
+          </Grid>
+        </a>
+      </Link>
+    </Column>
+  )
+}
+
+DesignKitCatalogItem.defaultProps = {
+  designKit: designKitPropTypes.isRequired
+}
```
Comment (francinelucca) [inline services/web-app/components/design-kit-catalog-item/design-kit-catalog-item.js:117]:
```suggestion
```

**Review by francinelucca — DISMISSED** [review]
Visuals LGTM, some corrections

File: services/web-app/components/framework-icon/framework-icon.js (hunk 14 above)
Comment (andreancardona) [reply services/web-app/components/framework-icon/framework-icon.js:44]: hehe whoops!

**Review by alisonjoseph — APPROVED** [review]
🚀 LGTM!

**Comment by kingtraceyj** [comment]
Just a few revisions

**Drop shadow on scroll**
Looks like the drop shadow is missing from the sticky search and filter bar. This is happening in my local environement on all catalogs but not happening in prod. Not sure what's going on.

image

Should be:
image

**Sort by alignment:**
This is also happening in other catalogs but also happening on prod. Sort by should align with the text above and below it. Should I open a bug issue for this?
image

**Tool icons:**
Tool icons are looking a little large. Max height and width should be 16px
image

**Update adobe xd tool icon:**
We should update the xd icon to the plex one
[16-adobeXD-plex.svg.zip](https://github.com/carbon-design-system/carbon-platform/files/9558840/16-adobeXD-plex.svg.zip)

**Comment by kingtraceyj** [comment]
**Tool icons**
Looks like they are 16px max. Design will need to further investigate the visual weight. Most of these icons will be available in the icon library soon so that might help with the issue. We can come back to this at a later time.

**Review by mattrosno — CHANGES_REQUESTED** [review]
@kingtraceyj @andreancardona is there a reason why the custom icons aren't #000? All other custom icons are true black, so then we can set color via `fill` CSS property.

Also, the Xd icon has a background that needs removed?
image

**Review by mattrosno — DISMISSED** [review]
Should kit name be duplicated up there or should that show the maintaining team's name? @kingtraceyj 

image

**Comment by kingtraceyj** [comment]
@mattrosno good catch. that should be the maintainer's team name.

File: services/web-app/components/design-kits-catalog/design-kits-catalog.js [hunk 24]
```diff
@@ -0,0 +1,187 @@
...
+      itemName={'design kit'}
+      availableFilters={availableFilters}
+      onFilter={filterDesignKits}
+      onUpdateFilter={updateFilter}
+      sortOptions={sortItems}
+      allowMultiView={false}
+    />
+  )
+}
+
```
Comment (francinelucca) [inline services/web-app/components/design-kits-catalog/design-kits-catalog.js:186]:
```suggestion

DesignKitsCatalog.propTypes = {
  /**
   * Collection of designKits to display.
   */
  designKits: PropTypes.arrayOf(designKitPropTypes).isRequired
}

```

**Review by francinelucca — DISMISSED** [review]
minor CR

**Comment by andreancardona** [comment]
> Should kit name be duplicated up there or should that show the maintaining team's name? @kingtraceyj
> 
> image

fixed

**Comment by andreancardona** [comment]
> fixed ✅

**Comment by kingtraceyj** [comment]
looking good from design!

design will follow up on the icon sizing once they icons are added to the icon library.

File: services/web-app/components/catalog-sort/catalog-sort.module.scss [hunk 25]
```diff
@@ -61,6 +61,7 @@
 
   // overide carbon color
   :global(.cds--label) {
+    padding-left: 0;
```
Comment (francinelucca) [inline services/web-app/components/catalog-sort/catalog-sort.module.scss:64]: Doesn't look right on mobile I think 🤔 
image

**Review by francinelucca — DISMISSED** [review]

**Review by francinelucca — APPROVED** [review]
💃 💃

File: services/web-app/utils/schema.js [hunk 26]
```diff
@@ -130,13 +150,23 @@ export const collapseAssetGroups = (asset, filter) => {
...
 }
 
+/**
+ * Gets the license from a designKit
+ * @param {import('@/typedefs').DesignKit} designKit
+ * @returns {string} License
+ */
+export const getDesignKitLicense = (designKit) => {
+  const license = licenses[designKit.license]?.name
+  return license
```
Comment (mattrosno) [inline services/web-app/utils/schema.js:167]: I'm surprised SonarCloud doesn't complain about this wanting:

```suggestion
  return licenses[designKit.license]?.name
```

Just commenting. Doesn't need changed IMO.

Comment (andreancardona) [reply services/web-app/utils/schema.js:167]: 🤷🏽‍♀️

File: services/web-app/utils/schema.js [hunk 27]
```diff
@@ -8,19 +8,20 @@
...
 import { tagsForCollection, tagsForType } from '@/data/tags'
-
-import { getSlug } from './slug'
+import { getSlug } from '@/utils/slug'
 /**
- * Defines the sort order of assets by status
- * @param {import('@/typedefs').Asset} assetA
- * @param {import('@/typedefs').Asset} assetB
+ * Defines the sort order of an element by status
+ * @param {import('@/typedefs').Element} elementA
```
Comment (mattrosno) [inline services/web-app/utils/schema.js:18]: Is Element in `typedefs.js`?

File: services/web-app/data/licenses.js [hunk 28]
```diff
@@ -0,0 +1,21 @@
...
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+export const licenses = {
+  'apache-2.0': {
+    name: 'Apache 2.0'
+  },
+  'ibm-internal': {
+    name: 'IBM Internal'
```
Comment (mattrosno) [inline services/web-app/data/licenses.js:13]: We might want this to be "IBM proprietary" per our OSS manager, but I'll create a separate issue to track this. Not urgent to do now.

Comment (mattrosno) [reply services/web-app/data/licenses.js:13]: Instead of "IBM internal".

File: services/web-app/utils/schema.js (hunk 27 above)
Comment (andreancardona) [reply services/web-app/utils/schema.js:18]: I'll update that

**Pull request merged**